crates/biome_js_syntax/src/expr_ext.rs

```diff
@@ -1258,3 +1263,27 @@ fn is_optional_chain(start: AnyJsExpression) -> bool {
 
     false
 }
+
+impl TsStringLiteralType {
+    /// Get the inner text of a string not including the quotes
+    ///
+    /// ## Examples
+    ///
+    /// ```
+    /// use biome_js_factory::make;
+    /// use biome_rowan::TriviaPieceKind;
+    ///
+    /// let string = make::ts_string_literal_type(make::js_string_literal("foo")
+    ///     .with_leading_trivia(vec![(TriviaPieceKind::Whitespace, " ")]));
+    /// assert_eq!(string.inner_string_text().unwrap().text(), "foo");
+    /// ```
+    pub fn inner_string_text(&self) -> SyntaxResult<TokenText> {
+        Ok(inner_string_text(&self.literal_token()?))
+    }
+}
+
+impl TsNumberLiteralType {
+    pub fn inner_string_text(&self) -> SyntaxResult<TokenText> {
+        Ok(inner_string_text(&self.literal_token()?))
```

Comment:
I am not sure it makes sense of calling/defining `inner_string_text` here because it si intended to retrieve a text between quotes ("string text"). If you need to retrieve a `TokenText`, you can use at call site the following code: `number_expr.value_token()?.token_text()`